cmd/agent/api/agent/agent.go

```diff
@@ -106,6 +106,12 @@ func makeFlare(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// remove the server_timeout for this connection, as generating a flare can
+	// take some time
+	conn := GetConnection(r)
+	_ = conn.SetDeadline(time.Time{})
+	_ = conn.SetWriteDeadline(time.Time{})
```

Comment:
I think you only want one or the other - not both:
```
	// SetDeadline sets the read and write deadlines associated
	// with the connection. It is equivalent to calling both
	// SetReadDeadline and SetWriteDeadline.
```

Reply:
Oh, huh, I copied this from the streaming-logs thing.  I'll fix in both.